### PR TITLE
Add versioneer to track package version automatically 

### PR DESCRIPTION
**Pull request by mikemhenry**
Fixes issue #803 

Also since I had to modify setup.py, I fixed the license and supported python versions, which fixes issue #788

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,33 +49,49 @@ def find_package_data(data_root, package_root):
...
-       'openmm.forcefielddir' : ['perses=perses:get_datadir'],
-       'console_scripts' : ['perses-relative = perses.app.setup_relative_calculation:run','perses-fah = perses.app.fah_generator:run']
-      },
-      ext_modules=extensions,
-      install_requires=[],
-      )
+setup(
+    name="perses",
+    author="Patrick Grinaway",
+    author_email="patrick.grinaway@choderalab.org",
```
Comment (jchodera) [inline setup.py:62]: We should update the authorship lines here. I could not find [a way to specify multiple authors](https://stackoverflow.com/questions/9999829/how-to-specify-multiple-authors-emails-in-setup-py), so how about I claim responsibility for now?
```python
    author="John D. Chodera",
    author_email="john.chodera@choderalab.org",
```
```

**Review by jchodera — APPROVED** [review]
Thanks so much! Please merge after updating the `author` fields.

File: setup.py (hunk 1 above)
Comment (mikemhenry) [reply setup.py:62]: Sounds good! I view scientific software authorship like it works with papers, last name got the funding and is the person to email if you have questions!

**Comment by jchodera** [comment]
> Sounds good! I view scientific software authorship like it works with papers, last name got the funding and is the person to email if you have questions!

Well, it's more for maintainership purposes right now: I'll still be the one users can bug even when folks move on!

**Pull request merged**